provisioning.go

```diff
@@ -388,6 +406,47 @@ func (prov *ProvisioningAPI) checkSessionAndReturnHandle(ctx context.Context, w
 	return handle
 }
 
+func (prov *ProvisioningAPI) WhoAmI(w http.ResponseWriter, r *http.Request) {
+	user := r.Context().Value(provisioningUserKey).(*User)
+	log := prov.log.With().
+		Str("action", "whoami").
+		Str("user_id", user.MXID.String()).
+		Logger()
+	log.Debug().Msg("getting whoami")
+
+	status := http.StatusOK
+	data := &WhoAmIResponse{
+		Permissions: int(user.PermissionLevel),
+		MXID:        user.MXID.String(),
+	}
+	if user.IsLoggedIn() {
+		puppet := user.bridge.GetPuppetBySignalID(user.SignalID)
+		if puppet == nil {
+			status = http.StatusInternalServerError
+			data.Signal = &WhoAmIResponseSignal{
+				Number: user.SignalUsername,
+				Ok:     false,
+				Error:  "Failed to get puppet by Signal ID",
+			}
+			log.Error().
+				Str("signal_id", user.SignalID.String()).
+				Msg(data.Signal.Error)
```

Comment:
This case doesn't really do anything, so it should just be removed. The `ok` boolean could also be removed or renamed. Instead there should be a boolean indicating if the client is connected to signal servers

Reply:
`ok` was returned in the signald version of the bridge, so IMO it should be kept around one way or another to preserve backwards compatibility.

I changed `ok` to contain the connection status; if you'd rather that be in a more aptly-named property (like `connected` or something), I'll change it.